src/ConnectedMode/QualityProfiles/QualityProfileDownloader.cs

```diff
@@ -94,19 +93,19 @@ public async Task<bool> UpdateAsync(BoundSonarQubeProject boundProject, IProgres
 
             EnsureProfilesExistForAllSupportedLanguages(boundProject);
 
+            var qpToUpdate = await outOfDateQualityProfileFinder.GetAsync(boundProject, cancellationToken);
+            
             var bindingConfigs = new List<IBindingConfig>();
 
-            var languageCount = languagesToBind.Count();
             int currentLanguage = 0;
+            var totalLanguages = languagesToBind.Count();
 
-            foreach (var language in languagesToBind)
+            foreach (var (language, qualityProfileInfo) in qpToUpdate)
             {
                 currentLanguage++;
 
                 var progressMessage = string.Format(BindingStrings.DownloadingQualityProfileProgressMessage, language.Name);
-                progress?.Report(new FixedStepsProgress(progressMessage, currentLanguage, languageCount));
-
-                var qualityProfileInfo = await TryDownloadQualityProfileAsync(boundProject, language, cancellationToken);
+                progress?.Report(new FixedStepsProgress(progressMessage, currentLanguage, totalLanguages));
```

Comment:
The last parameter in `progress.Report` is the total number of steps that will be performed. We're only processing the out of date QPs now - shouldn't the total be `qpToUpdate.Count()`?

Reply:
Yes, but I got warnings about double enumeration, so I decided to just take the worst case scenario in terms of number of steps. Sure, sometimes it will stop before that, but nothing would break.

Reply:
> I got warnings about double enumeration

What do you mean? Couldn't you just store the value once in line 101?


BTW this is the sort of thing that you should call out explicitly in code review comments, and not wait to see if the reviewer notices it. It's not obvious, easy to miss, and means the UX won't be what the user expects. You've made a clear decision to handle the problem in a particular way, so call it out in a review comment.